src/Process/PointCloudProcessing.js

```diff
@@ -0,0 +1,291 @@
+                    elt.obj.material.uniforms.size.value = layer.pointSize;
+                } else if (!elt.promise) {
+                    elt.promise = context.scheduler.execute({
+                        layer,
+                        requester: elt,
+                        view: context.view,
+                        priority: 1.0 / elt.name.length, // surfaceOnScreen,
+                        redraw: true,
+                        isLeaf: elt.childrenBitField == 0,
+                    }).then((pts) => {
+                        elt.obj = pts;
+                        // store tightbbox to avoid ping-pong (bbox = larger => visible, tight => invisible)
+                        elt.tightbbox = pts.tightbbox;
+                        pts.geometry.setDrawRange(
+                            0, shouldBeLoaded * pts.geometry.attributes.position.count);
+
+                        // make sure to add it here, otherwise it might never
+                        // be added nor cleaned
+                        layer.group.add(elt.obj);
+
+                        elt.obj.owner = elt;
+                        elt.promise = null;
+                    });
+                }
+            }
+        } else if (shouldBeLoaded < 0) {
+            // not visible
+            if (!elt.notVisibleSince) {
+                elt.notVisibleSince = Date.now();
+            }
+        }
+
+        if (shouldBeLoaded >= 0.9 && elt.children && elt.children.length) {
+            return elt.children;
+        }
+        return undefined;
+    },
+
+    postUpdate(context, layer) {
+        if (!layer.group) {
+            return;
+        }
+
+        if (layer.counters.displayedCount > layer.pointBudget) {
+            const reduction = layer.pointBudget / layer.counters.displayedCount;
+            for (const pts of layer.group.children) {
+                if (pts.material.visible) {
+                    pts.geometry.setDrawRange(0, pts.geometry.drawRange.count * reduction);
+                }
+            }
+            layer.counters.displayedCount *= reduction;
+        }
+
+        const now = Date.now();
+
+        for (let i = 0; i < layer.group.children.length; i++) {
+            const obj = layer.group.children[i];
+            if (!obj.material.visible && (now - obj.owner.notVisibleSince) > 10000) {
+                // remove from group
+                layer.group.children.splice(i, 1);
+
+                obj.material.dispose();
+                obj.geometry.dispose();
+                obj.material = null;
+                obj.geometry = null;
+                obj.owner.obj = null;
+
+                if (__DEBUG__) {
+                    if (obj.boxHelper) {
+                        for (let j = 0; j < layer.bboxes.children.length; j++) {
+                            if (layer.bboxes.children[j].id == obj.boxHelper.id) {
+                                layer.bboxes.children.splice(j, 1);
```

Comment:
By using splice with a forward for loop, you're going to jump above some elements right?